ingestion/operators/docker/test_connection.py

```diff
@@ -54,7 +57,10 @@ def main():
     test_connection_fn = get_test_connection_fn(
         test_service_connection.connection.config
     )
-    test_connection_fn(connection)
+    results: Optional[TestConnectionResult] = test_connection_fn(connection)
+    if results:
+        with open("/tmp/test_connection_results.json", "w", encoding=UTF_8) as file:
```

Comment:
if there are multiple test connections going on , wouldn't this override the file with different connection results.
lets say there are 2 users logged in and each with their own sessions and simultaneously adding services can get unwanted results.
Also the case where we write a test_connection_results.json from previous test connection, current test connection fails , the polling of this file can return a stale result from previous file
IMO, storing these results in a file doesn't look like a good idea .
Might as well expand the connection on the server side to have test connection and write the results back there instead in a file

Reply:
these will happen running on an ephemeral pod. When we call the test_connection from Argo, the workflow will execute on a pod, write locally, and then expose the results via the output param read from the file: https://argoproj.github.io/argo-workflows/walk-through/output-parameters/